test/exe/win32_scm_test.cc

```diff
@@ -0,0 +1,168 @@
+#include <cstdio>
+#include <fstream>
+
+#include "test/test_common/environment.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+#ifdef WIN32
+class ScmTest : public testing::TestWithParam<Network::Address::IpVersion> {
```

Comment:
Test coverage so far looks good, seems like you've got the main flows covered (start, stop, exit on error, etc.) 👍

Reply:
coverage of the logic around running as a service or console app should be covered by existing tests

Reply:
The big question that I need the CI is if we can run these tests on CI because they require you to be admin. Locally from an msys2 Windows that is running as admin the test is hanging. But from Admin cmd just running the executable the test passes